dotCMS/src/main/java/com/dotcms/rest/api/v1/index/ESIndexResource.java

```diff
@@ -67,392 +68,384 @@
+        api.optimize(indices);
+        final String message = Try.of(()-> LanguageUtil.get(APILocator.getCompanyAPI().getDefaultCompany(),"message.cmsmaintenance.cache.indexoptimized")).get();
+        sendAdminMessage(message, MessageSeverity.INFO,init.getUser(), 0);
+        return Response.ok().build();
+    }
 
-                        Logger.error(this, "Exception trying to restore index snapshot:" + e.getMessage(), e);
-                        return ResponseUtil.mapExceptionResponse(e);
-                    },
-                    asyncResponse);
+    @CloseDBIfOpened
+    @DELETE
+    @JSONP
+    @NoCache
+    @Path("/cache")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response flushIndiciesCache(@Context final HttpServletRequest request, @Context final HttpServletResponse response) {
+        final InitDataObject init = auth(request, response);
+        final ContentletIndexAPI api = APILocator.getContentletIndexAPI();
+        final List<String> indices = api.listDotCMSIndices();
+        final Map<String, Integer> data = APILocator.getESIndexAPI().flushCaches(indices);
+        String message = Try.of(()->LanguageUtil.get(APILocator.getCompanyAPI().getDefaultCompany(),"maintenance.index.cache.flush.message")).get();
+        message=message.replace("{0}", String.valueOf(data.get("successfulShards")));
+        message=message.replace("{1}", String.valueOf(data.get("failedShards")));
+        sendAdminMessage(message, MessageSeverity.INFO, init.getUser(),5000);
+        return Response.ok(new ResponseEntityView(data)).build();
 
-        } catch (Exception e) {
-            Logger.error(this.getClass(),"Exception trying to restore index snapshot: " + e.getMessage(), e);
-            asyncResponse.resume(ResponseUtil.mapExceptionResponse(e));
-        }
     }
 
 
-    /**
-     * @deprecated  As of 2016-10-12, replaced by {@link #snapshotIndex(HttpServletRequest,HttpServletResponse ,String)}
-     */
+    @Deprecated
     @PUT
     @Path("/create/{params:.*}")
     @Produces("text/plain")
     public Response createIndex(@Context HttpServletRequest httpServletRequest, @Context final HttpServletResponse httpServletResponse, @PathParam("params") String params) {
         try {
-            InitDataObject init=auth(params,httpServletRequest, httpServletResponse);
+            InitDataObject init=auth(httpServletRequest, httpServletResponse);
 
             int shards=Integer.parseInt(init.getParamsMap().get("shards"));
             boolean live = init.getParamsMap().containsKey("live") ? Boolean.parseBoolean(init.getParamsMap().get("live")) : false;
             String indexName = init.getParamsMap().get("index");
 
-            indexName = create(indexName, shards, live);
+            if(indexName == null)
+                indexName=ContentletIndexAPIImpl.timestampFormatter.format(new Date());
+            indexName = (live) ? "live_" + indexName : "working_" + indexName;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Prefer StringBuilder (non-synchronized) or StringBuffer (synchronized) over += for concatenating strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)